MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -10,251 +13,209 @@ namespace MediaBrowser.Common.Net
         {
-            if (source == null)
+            if (string.IsNullOrWhiteSpace(host))
             {
-                throw new ArgumentNullException(nameof(source));
+                addresses = Array.Empty<IPAddress>();
+                return false;
             }
 
-            Collection<IPObject> res = new Collection<IPObject>();
+            host = host.Trim();
 
-            foreach (IPObject i in source)
+            // See if it's an IPv6 with port address e.g. [::1] or [::1]:120.
+            if (host[0] == '[')
             {
-                if (i is IPNetAddress nw)
-                {
-                    // Add the subnet calculated from the interface address/mask.
-                    var na = nw.NetworkAddress;
-                    na.Tag = i.Tag;
-                    res.AddItem(na);
-                }
-                else if (i is IPHost ipHost)
+                int i = host.IndexOf(']', StringComparison.Ordinal);
+                if (i != -1)
                 {
-                    // Flatten out IPHost and add all its ip addresses.
-                    foreach (var addr in ipHost.GetAddresses())
-                    {
-                        IPNetAddress host = new IPNetAddress(addr)
-                        {
-                            Tag = i.Tag
-                        };
-
-                        res.AddItem(host);
-                    }
+                    return TryParseHost(host.Remove(i)[1..], out addresses);
                 }
-            }
-
-            return res;
-        }
 
-        /// <summary>
-        /// Excludes all the items from this list that are found in excludeList.
-        /// </summary>
-        /// <param name="source">The <see cref="Collection{IPObject}"/>.</param>
-        /// <param name="excludeList">Items to exclude.</param>
-        /// <param name="isNetwork">Collection is a network collection.</param>
-        /// <returns>A new collection, with the items excluded.</returns>
-        public static Collection<IPObject> Exclude(this Collection<IPObject> source, Collection<IPObject> excludeList, bool isNetwork)
-        {
-            if (source.Count == 0 || excludeList == null)
-            {
-                return new Collection<IPObject>(source);
+                addresses = Array.Empty<IPAddress>();
+                return false;
             }
 
-            Collection<IPObject> results = new Collection<IPObject>();
+            var hosts = host.Split(':');
 
-            bool found;
-            foreach (var outer in source)
+            if (hosts.Length <= 2)
             {
-                found = false;
+                // Use regular expression as CheckHostName isn't RFC5892 compliant.
+                // Modified from gSkinner's expression at https://stackoverflow.com/questions/11809631/fully-qualified-domain-name-validation
+                string pattern = @"(?im)^(?!:\/\/)(?=.{1,255}$)((.{1,63}\.){0,127}(?![0-9]*$)[a-z0-9-]+\.?)(:(\d){1,5}){0,1}$";
```

Comment:
Regex should be created once (eg `private static readonly _fqdnRegex = new Regex($pattern);`